crates/cargo-gpu/src/metadata.rs

```diff
@@ -85,74 +66,64 @@ impl Metadata {
     /// Convert a `Cargo.toml` to JSON
     //
     // TODO: reuse for getting the default `rust-gpu` source and toolchain.
-    fn get_cargo_toml_as_json(path: &std::path::PathBuf) -> anyhow::Result<serde_json::Value> {
-        let cargo_toml_path = path.join("Cargo.toml");
-        if !cargo_toml_path.exists() {
-            anyhow::bail!("{path:?} must be a shader crate directory");
-        }
-
-        log::debug!("Querying Cargo metadata for {cargo_toml_path:?}");
-        let output_cargo = std::process::Command::new("cargo")
-            .args([
-                "metadata",
-                "--no-deps",
-                "--manifest-path",
-                cargo_toml_path.display().to_string().as_ref(),
-            ])
-            .output()?;
-        anyhow::ensure!(
-            output_cargo.status.success(),
-            "could not run `cargo metadata` on {cargo_toml_path:?}"
-        );
-
-        Ok(serde_json::from_slice(&output_cargo.stdout)?)
+    fn get_cargo_toml_as_json(
+        path: &std::path::PathBuf,
+    ) -> anyhow::Result<cargo_metadata::Metadata> {
+        Ok(MetadataCommand::new().current_dir(path).exec()?)
     }
 
     /// Get any `rust-gpu` metadata set in the root workspace `Cargo.toml`
-    fn get_workspace_metadata(json: &serde_json::Value) -> serde_json::Value {
-        let empty_json_object = serde_json::json!({});
-        let mut metadata = json
-            .pointer("/metadata/rust-gpu")
-            .unwrap_or(&empty_json_object)
-            .clone();
-
-        Self::keys_to_snake_case(&mut metadata);
-        metadata.clone()
+    fn get_workspace_metadata(metadata: &cargo_metadata::Metadata) -> Value {
```

Comment:
Could `get_rust_gpu_from_metadata` just be called directly? Then we could delete this function.